core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -444,6 +444,21 @@ private char[] getKeyStorePasswordAsCharArray(String keyStoreName) {
         return decryptPassword(encryptedPassword);
     }
 
+    /**
+     * Get the private key password for the given key store name.
+     *
+     * @param keyStoreName key store name.
+     * @return Private key password as a character array/
+     */
+    public char[] getPrivateKeyPassword(String keyStoreName) throws CarbonException {
+
+        if (KeyStoreUtil.isPrimaryStore(keyStoreName)) {
+            return getPrimaryPrivateKeyPasssword().toCharArray();
+        }
+        String encryptedPassword = keyStorePersistenceManager.getEncryptedKeyStorePassword(keyStoreName, tenantId);
```

Comment:
Why are we creating a separate method instead if making the existing method public?

Reply:
The existing method used to only retrieve the private key pwd of the the tenant not for the super tenant

Reply:
Either we can call the private method from here to avoid code duclication or remove the private method completely.

Reply:
Fixed with: https://github.com/wso2/carbon-kernel/compare/0111c22e9e2f59865964900e1da064d4c1faff25..f78579025f0e61c612a9f871a0962fc6c398d81a